apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  TestAnalysisQuery,
+  TestAnalysisQueryVariables,
+} from "gql/generated/types";
+import { TEST_ANALYSIS } from "gql/queries";
+import FailedTestGroup from "./FailedTestGroup";
+import {
+  countTotalTests,
+  filterGroupedTests,
+  getAllBuildVariants,
+  getAllTaskStatuses,
+  groupTestsByName,
+} from "./utils";
+
+const { green } = palette;
+interface TestAnalysisProps {
+  versionId: string;
+}
+const TestAnalysis: React.FC<TestAnalysisProps> = ({ versionId }) => {
+  const [selectedTaskStatuses, setSelectedTaskStatuses] = useState<string[]>(
+    [],
+  );
+  const [selectedBuildVariants, setSelectedBuildVariants] = useState<string[]>(
+    [],
+  );
+  const [searchValue, setSearchValue] = useState<string>("");
+  const { data, loading } = useQuery<
+    TestAnalysisQuery,
+    TestAnalysisQueryVariables
+  >(TEST_ANALYSIS, {
+    variables: {
+      versionId,
+      options: {
+        statuses: failedTaskStatuses,
+      },
+      opts: {
+        statuses: ["fail"],
+      },
+    },
+  });
+
+  const groupedTestsMap = groupTestsByName(
+    data ? data?.version?.tasks?.data : [],
+  );
+
+  const buildVariants = getAllBuildVariants(groupedTestsMap);
+  const taskStatuses = getAllTaskStatuses(groupedTestsMap);
+
+  const filteredGroupedTestsMap = filterGroupedTests(
+    groupedTestsMap,
+    searchValue,
+    selectedTaskStatuses,
+    selectedBuildVariants,
+  );
+  const groupedTestsMapEntries = Array.from(
+    filteredGroupedTestsMap.entries(),
+  ).sort((a, b) => b[1].length - a[1].length);
+
+  const numberOfTestsThatFailedOnMoreThanOneTask =
+    groupedTestsMapEntries.filter(([, tasks]) => tasks.length > 1).length;
+
+  const totalTestCount = countTotalTests(groupedTestsMap);
+  const totalFilteredTestCount = countTotalTests(filteredGroupedTestsMap);
+  const hasFilters =
+    selectedTaskStatuses.length > 0 ||
+    selectedBuildVariants.length > 0 ||
+    searchValue.length > 0;
+
+  const hasMatchingResults = totalFilteredTestCount > 0;
+  const hasResults = data && totalTestCount > 0;
+  return (
+    <div>
+      {loading ? (
+        <ListSkeleton />
+      ) : (
+        <div>
+          <Title>
+            {numberOfTestsThatFailedOnMoreThanOneTask}{" "}
+            {pluralize("test", numberOfTestsThatFailedOnMoreThanOneTask)} failed
+            across more than one task
+          </Title>
+          <Body>
+            This page shows tests that failed across more than one task. If a
+            test failed on multiple tasks, it may indicate a flaky test or a
+            larger issue. Click on the test name to see more details.
+          </Body>
+          <FilterContainer>
+            <div>
+              <LabelWrapper>
+                <Label
+                  htmlFor="test-failure-search"
+                  id="test-failure-search-label"
+                >
+                  Search Test Failures
+                </Label>
+              </LabelWrapper>
+              <SearchInput
```

Comment:
[nit] It would be nice if the focus overflow wasn't cut off
<img width="393" alt="image" src="https://github.com/user-attachments/assets/fc9293ec-66be-4dc5-ae8c-050bf8d1788e">